File: abstractions/dotnet/src/ApiException.cs
// ------------------------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
// ------------------------------------------------------------------------------
using System;

namespace Microsoft.Kiota.Abstractions;

/// <summary>
/// Parent type for exceptions thrown by the client when receiving failed responses to its requests.
/// </summary>
public class ApiException : Exception
{
    /// <inheritdoc/>
    public ApiException(): base()
    {
        
    }
    /// <inheritdoc/>
    public ApiException(string message) : base(message)
    {
    }
    /// <inheritdoc/>
    public ApiException(string message, Exception innerException) : base(message, innerException)
    {
    }
}

File: abstractions/php/src/ApiException.php
<?php

namespace Microsoft\Kiota\Abstractions;
use Exception;

class ApiException extends Exception
{
    /**
     * @param string $message
     * @param int $code
     * @param Exception|null $innerException
     */
    public function __construct(string $message = "", int $code = 0, ?Exception $innerException = null) {
        parent::__construct($message, $code, $innerException);
    }
}